### ISSUE TITLE
fetch doesn't stream

### ISSUE THREAD

**Issue by janwilmake** [oven-sh/bun#4132]
### What version of Bun is running?

0.7.3

### What platform is your computer?

Darwin

### What steps can reproduce the bug?

I just made this script to stream OpenAI tokens back to my node server...

```js
import { ChatCompletionRequestMessage } from "openai";
import { ChatCompletionConfig } from "./types";

export function notEmpty<TValue>(
  value: TValue | null | undefined
): value is TValue {
  return value !== null && value !== undefined;
}

export const streamChatCompletion = async (
  messages: ChatCompletionRequestMessage[],
  config: ChatCompletionConfig,
  onData: (text: string) => void
) => {
  if (!config.credentials.openAiToken) {
    return { isSuccessful: false, message: "No token" };
  }

  const controller = new AbortController();

  try {
    const response = await fetch("https://api.openai.com/v1/chat/completions", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${config.credentials.openAiToken}`,
      },

      body: JSON.stringify({
        stream: true,
        model: "gpt-3.5-turbo",
        messages,
        ...config.openaiConfig,
      }),

      signal: controller.signal,
    });

    const reader = response.body?.getReader();
    const decoder = new TextDecoder("utf-8");

    if (!reader) {
      return { isSuccessful: false, message: "No stream" };
    }

    while (true) {
      const { done, value } = await reader.read();
      if (done) {
        break;
      }

      const chunk = decoder.decode(value);
      //   console.log({ chunk });
      const lines = chunk.split("\n");
      const text = lines
        .map((line) => line.replace(/^data: /, "").trim()) // Remove the "data: " prefix
        .filter((line) => line !== "" && line !== "[DONE]") // Remove empty lines and "[DONE]"
        .map((line) => JSON.parse(line))
        .map((json) => json.choices?.[0].delta.content)
        .filter(notEmpty)
        .join(""); // Parse the JSON string

      onData(text);
    }
  } catch (error) {
    console.error("Error fetching data:", error);

    return { isSuccessful: false, message: "Streaming fetch went wrong" };
  }
};

streamChatCompletion(
  [
    { role: "system", content: "You are a helpful assistant" },
    { role: "user", content: "Hi, what is love?" },
  ],
  {
    credentials: {
      openAiToken: "TOKEN",
    },
    openaiConfig: {},
  },
  (text) => process.stdout.write(text)
);
```

### What is the expected behavior?

In node.js it streams the result to the console, token by token.

### What do you see instead?

In bun, it crashes and doesn't stream

### Additional information

Seems fetch isn't fully implemented (https://bun.sh/docs/runtime/nodejs-apis)

**Comment by Jarred-Sumner** [comment]
yep, @cirospaciari is working on it in #4127

**Comment by birkskyum** [comment]
Closed by bun 0.8

**Comment by janwilmake** [comment]
I can confirm the stream is indeed working, also as endpoint as per the example in the 0.8 blog (https://bun.sh/blog/bun-v0.8.0)
